runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/manager/SdkHarnessManager.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.fnexecution.manager;
+
+import org.apache.beam.model.fnexecution.v1.ProvisionApi;
+import org.apache.beam.runners.core.construction.graph.ExecutableStage;
+import org.apache.beam.runners.fnexecution.artifact.ArtifactSource;
+import org.apache.beam.runners.fnexecution.control.RemoteBundle;
+import org.apache.beam.runners.fnexecution.state.StateRequestHandler;
+
+/**
+ * A manager of resources related to the SDK Harness, capable of providing RemoteBundles to runner
+ * operators.
+ *
+ * <p>In order to provide a simple interface to runner operators, the SdkHarnessManager is
+ * responsible for owning and managing the lifetimes of resources such as RPC servers and remote
+ * environments. It is responsible for both instantiation and cleanup of these resources.  Since all
+ * managed resources are owned by the SdkHarnessManager, it is responsible for cleaning them up when
+ * its close function is called.
+ */
+public interface SdkHarnessManager extends AutoCloseable {
+  /**
+   * Get a new {@link RemoteBundle bundle} for processing the data in an executable stage.
+   *
+   * <p>If necessary, this blocks while provisioning the remote resources necessary to support
+   * bundle processing.
+   */
+  <InputT> RemoteBundle<InputT> getBundle(
```

Comment:
This feels like the wrong high-level interface to use everywhere - some runners will have persistent ArtifactSources and Provisioning information, at minimum, and should not need to respecify a server-shared one every time.

Reply:
That's a good point.  I'll decouple the registration of operator resources from bundle retrieval.

Reply:
Ok.  commented on wrong thread initially.  This is the file that contains a fix.

Reply:
I did not notice `jobId` - that seems definitely wrong - if you have a shared `SdkHarnessManager`, it should definitely have only a single associated Job ID.

My general mental shape of things is roughly the following:
The interface that the processor tends to use is basically `RemoteBundle<?> getBundle(ExecutableStage stage, StateRequestHandler stateHandler)` (as I understand it, the StateRequestHandler isn't bound to a service - but is bound to some external scope of the bundle, as understood by a runner, so should be passed for each bundle)

For flink, or other runners that have , that may have to be wrapped in something which provides a `HarnessManager getHarnessManager(jobId, ArtifactSource, ... [job provisioning basics])`. For runners with single-job and persistent artifact sources, they can use the simpler interface only.

I don't see any strongly compelling reason to include them all in the same interface - unless I'm suggesting duplicating an interface that we already have.